pkg/scheduler/internal/cache/cache.go

```diff
@@ -525,8 +525,7 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods[key]:
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.Errorf("Pod %v updated on a different node than previously added to.", key)
-			klog.Fatalf("Schedulercache is corrupted and can badly affect scheduling decisions")
+			klog.ErrorS(fmt.Errorf("schedulercache is corrupted and can badly affect scheduling decisions"), "Pod updated on a different node than previously added to", "pod", key)
```

Comment:
Maybe missed os.Exit(1)

Reply:
Yes, It was poor thinking-through of a code change. I will fix it.